test/sharness/t0252-files-gc.sh

```diff
@@ -42,4 +43,34 @@ test_expect_success "gc okay after adding incomplete node" '
   ipfs refs $ADIR_HASH
 '
 
+MYDIR_HASH=QmRLrxQTdmLZNfDJxFQ9uFrgtj4s6cZU9GzTxCq2YLev24
+HELLO_HASH=QmeV1kwh3333bsnT6YRfdCRrSgUPngKmAhhTa4RrqYPbKT
+
+test_expect_success "add directory with direct pin" '
+  mkdir mydir/ &&
+  echo "hello world!" > mydir/hello.txt &&
+  ipfs add --pin=false -r mydir &&
+  ipfs pin add --recursive=false $MYDIR_HASH &&
+  ipfs cat $HELLO_HASH
```

Comment:
if something changes, the hashes may no longer match watch you got. if it's important that it matches, check it explicitly. if it is not important that it matches, grab the hashes directly from the output of `ipfs add` with something like:

```sh
HELLO_HASH=$(ipfs add --pin=false -q -r mydir/hello.txt | tail -n1)
MYDIR_HASH=$(ipfs add --pin=false -q -r mydir | tail -n1)
```